src/_pytest/config/__init__.py

```diff
@@ -617,20 +622,36 @@ class Config:
 
     :ivar argparse.Namespace option: access to command line option as attributes.
 
-    :ivar invocation_args: list of command-line arguments as passed to pytest.main()
+    :ivar InvocationParams invocation_params:
 
-    :ivar invocation_plugins: list of extra plugins passed to pytest.main(), might be None
+        Object containing the parameters regarding the ``pytest.main``
+        invocation.
 
-    :ivar py.path.local invocation_dir: directory where pytest.main() was invoked from
+        Contains the followinig read-only attributes:
+
+        * ``args``: list of command-line arguments as passed to ``pytest.main()``.
+        * ``plugins``: list of extra plugins, might be None
+        * ``dir``: directory where ``pytest.main()`` was invoked from.
     """
 
-    def __init__(self, pluginmanager):
+    @attr.s(frozen=True)
+    class InvocationParams:
+        """Holds parameters passed during ``pytest.main()``"""
+
+        args = attr.ib()
+        plugins = attr.ib()
+        dir = attr.ib()
+
+    def __init__(self, pluginmanager, *, invocation_params=None):
         from .argparsing import Parser, FILE_OR_DIR
 
+        if invocation_params is None:
```

Comment:
we might want to experiment with making this mandatory as a follow-up to this pr

Reply:
extra note, there also is a clarification required for how to deal with PYTEST_ADDOPTS

Reply:
> extra note, there also is a clarification required for how to deal with PYTEST_ADDOPTS

What do you mean? Currently there's nothing explicit, the user needs to take that into account explicitly. 

We might add `addopts_env` to `InvocationParam` though.

Reply:
@nicoddemus i mean there should be a explicit specification about how and when it is integrated, we shouldn't make i configurable in a new way

Reply:
> we shouldn't make i configurable in a new way

What do you mean? I'm not touching `PYTEST_ADDOPTS` at all here, so I don't understand why this PR is "making it configurable in a new way"... 🤔 

Perhaps you meant for me to write some docs about when the user is expected to include `PYTEST_ADDOPTS` into the command line?

Reply:
no - we need to clarify that it has a effect and is a "implicit parameter" and that it will never be configurable fore this kind of invocation from the "python side" so users with a need to control it need to control it themselves

Reply:
Got it, done. Let me know if you would like some tweak on the text.

I will try to see if it is possible to initialize the plugins inside the config object as a follow up.